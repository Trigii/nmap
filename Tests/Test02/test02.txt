Authors: Javier Sande Ríos; Tristán Vaquero Povedano

Description: Scan host with a valid and not blacklisted self-signed certificate with weak algorithms (MD5) and short validity (less that one month).

Input:
	* list: blacklist.csv
	* dns-list: dns.csv

Output:
	* Information about the certificate
	* Self-signed: True
	* Verified: True
	* Validity: True
	* Valid names: True
	* Blacklisted: False
	* DNS resolved: True
	* DNS: IP in valid range
	* Warnings:
		* Weak algorithms
		* Short validity

Command: $ nmap -Pn -p443 10.0.3.149 --script untrustedX509certs.nse --script-args list=blacklist.csv dns-list=dns.csv

