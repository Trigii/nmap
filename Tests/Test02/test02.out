# Nmap 7.91 scan initiated Mon Nov 13 11:48:51 2023 as: nmap -Pn -p443 --script untrustedX509certs.nse --script-args list=blacklist.csv -v -oN test02.out 10.0.3.149 dns-file=dns.csv
Failed to resolve "dns-file=dns.csv".
Failed to resolve "dns-file=dns.csv".
Nmap scan report for 10.0.3.149
Host is up (0.00049s latency).

PORT    STATE SERVICE
443/tcp open  https
| untrustedX509certs: Subject: commonName=10.0.3.149/organizationName=UC3M/stateOrProvinceName=MADRID/countryName=ES/organizationalUnitName=UC3M Cyber/localityName=GETAFE
| Subject Alternative Name: DNS:cyber.uc3m.es, DNS:10.0.3.149
| Issuer: commonName=10.0.3.149/organizationName=UC3M/stateOrProvinceName=MADRID/countryName=ES/organizationalUnitName=UC3M Cyber/localityName=GETAFE
| Public Key type: rsa
| Public Key bits: 4096
| Signature Algorithm: md5WithRSAEncryption
| Not valid before: 2023-11-13T15:47:53
| Not valid after:  2023-12-03T15:47:53
| MD5:   05fc 677e 0780 4438 2dd9 959b 3253 3741
| SHA-1: be4b bbf5 75b2 67bb 9947 193f 7307 733c bbf1 9ee2
| -----BEGIN CERTIFICATE-----
| MIIF1zCCA7+gAwIBAgIUZy1pD+EaBh+T+yBOn1MOb/+wAtkwDQYJKoZIhvcNAQEE
| BQAwaDELMAkGA1UEBhMCRVMxDzANBgNVBAgMBk1BRFJJRDEPMA0GA1UEBwwGR0VU
| QUZFMQ0wCwYDVQQKDARVQzNNMRMwEQYDVQQLDApVQzNNIEN5YmVyMRMwEQYDVQQD
| DAoxMC4wLjMuMTQ5MB4XDTIzMTExMzE1NDc1M1oXDTIzMTIwMzE1NDc1M1owaDEL
| MAkGA1UEBhMCRVMxDzANBgNVBAgMBk1BRFJJRDEPMA0GA1UEBwwGR0VUQUZFMQ0w
| CwYDVQQKDARVQzNNMRMwEQYDVQQLDApVQzNNIEN5YmVyMRMwEQYDVQQDDAoxMC4w
| LjMuMTQ5MIICIjANBgkqhkiG9w0BAQEFAAOCAg8AMIICCgKCAgEAvUS+jVq28lC1
| r26kh3P5zO9HnEKrnW95YSzz6eRTUqopAfHED07e4QHLr8Eco32iC35GEpmXjaQD
| aEeU3w4Ia1WlzzePJmdRf/NtFsHW9PbDKHBuFY0Y43TWZU/pU5gXA4pLkFVJNEQk
| D8ns2jS4dgmWg4mlKLmobTM8CDKX0OzivaTCtTQje1cjPhRDpwtr7sB0cvmDF9Ul
| O3eKEDP0xcFYzv42Le4uUgp/SqKmsOylXDIcJolT8hQV/rFGYevC0kdpGV2uS8+q
| KPNZBTA7bHNRw6KeWkidwv0cv1iWtOLKYwR/H1tlKLDYqHWKlOcVisz+vRRobZKl
| YEMmRKv4leG6PJL/Abhe72aCFuhjtdfvBfZ10LFPwzlnvfPzC35S+jjuxVh+39Nb
| fG3BdsMRDnAXbNgmOjm2zoJvRoYk8jhMoDG7oJW620xScuDHRjJwZcj1gSPEn7Ib
| I1LBagbQSFnNmqAjMuAZEcwt4OG+gitP7ZX9RL5wj7E1OkF1jkOV6k0T+5Xix0Mr
| xj/YiGL8FoEKNXRGETbZmxf58Hn7w/X0xNzRfLmDp+uyG/zN26uZaZ2EJODbptPS
| gGmlNZY7OL39LYEHuQ2/HjibOabAuCfO9i/vNAfwM7e+C6bauWcHKE6R9IdWHyn1
| ziIzcXZL/cW9vUB1m1852wOBdTtQ04cCAwEAAaN5MHcwJAYDVR0RBB0wG4INY3li
| ZXIudWMzbS5lc4IKMTAuMC4zLjE0OTAdBgNVHQ4EFgQUC8fXhMbAX+g2cn7QzRJ3
| TqP4cjwwHwYDVR0jBBgwFoAUC8fXhMbAX+g2cn7QzRJ3TqP4cjwwDwYDVR0TAQH/
| BAUwAwEB/zANBgkqhkiG9w0BAQQFAAOCAgEAX45VNeIy7k6XtY1u3KWPC8mNotKK
| ryTnfY3ccO6VboFunDnm2bK0fAIrn7vafcJ5HbSgBrixpmVWJJ5Vw7mDn3mZK9YU
| I2u/hAUcZwKGlCRP5dc6NJ7ASGkHEp/Jxlipwu1dJedaKRzZVkhYgYoxUVH/IVgP
| 8BQCDOxapRr62sG0BDuWSS/CaeI7sl+7kaZcpiN39kweTwvj026NRBjZ49Z98Kiu
| eDgugAj5rP8+eId3b1BzJYaLS7/1MtE3SikddoLgkN9PZ6MmyPb992PPgUM/1YyO
| RvmE8lkBQRjz+l/wif08CFFQvUrKXukbWKVwGX8WXIpZhh9p4oLEmIONamIl+jjr
| gYnCsaoojYeTZW9uxZEiXe69DQ/yrYBGBBVLv2hOPk7TZdpZjQH8i9r2hPDFQCLK
| mEwwdXBcNnUHyiKyVKD5EAXeLnhrMj+glK+DpZq9VXRbrwcDaGXYBA3lEom8bKan
| 3vyeP4EBn7s9UGfgP9PpVXwGvybxEULLlUIbhoD7nztQmAQXp9noeqgzQuD9L+C5
| pSpGVSkMBYo8QEWh4xd1a4/S8ac7Q2Vv7zSJM+XkYhE/0n0CBH6/OUlbcsM3OkMS
| rEq5vsyOZ5TxwDlc2j3pE1RPHdROgGlycj+V1ZqrFPB+dVrHGp8BGBwPXqoI9q1a
| M+aGVieWDHhZzKo=
| -----END CERTIFICATE-----
| 
| --------------------------------
| CERTIFICATE SECURITY REPORT
| --------------------------------
| Self-signed: True
| Certificate verified
| Validity: Certificate is VALID
| Certificate Subject Alternative Names contains hostname
| Blacklisted: False
| DNS resolved: True
| DNS: server ip is in a VALID RANGE
| WARNING: Short validity (less than one month)
|_WARNING: signature algorithm is not strong (It should be RSA or DSA or ECDSA with SHA256 or stronger).

Failed to resolve "dns-file=dns.csv".
Read data files from: /usr/bin/../share/nmap
# Nmap done at Mon Nov 13 11:49:05 2023 -- 1 IP address (1 host up) scanned in 13.28 seconds
